packages/react-router/modules/matchPath.js

```diff
@@ -1,5 +1,59 @@
 import pathToRegexp from "path-to-regexp";
 
+function isAbsolute(pathname) {
+  return pathname.charAt(0) === "/";
+}
+
+function ensureTrailingSlash(pathname) {
+  return hasTrailingSlash(pathname) ? pathname : pathname + "/";
+}
+
+function hasTrailingSlash(pathname) {
+  return pathname.charAt(pathname.length - 1) === "/";
+}
+
+function isMalformed(pathname) {
+  return pathname.slice(0, 3) === "...";
+}
+
+function refersToParentSegment(pathname) {
+  return pathname.slice(0, 3) === "../";
+}
+
+function refersToCurrentSegment(pathname) {
+  return pathname.slice(0, 2) === "./";
+}
+
+function resolvePath(pathname, base) {
```

Comment:
I don't quite like how spread the code is for resolvePath().  Encapsulating all the criteria in separate functions makes for uneasy reading if you need to debug or want to understand the minute details of the implementation to enhance it. And non of these functions are reused (or particularly reuseable), so DRY doesn't apply.

Reply:
The alternative solution would be to add comments. In my choice I prefer functions with calling names over comments that could be outdated even without noticing.